sites/partners/src/listings/PaperListingForm/sections/Preferences.tsx

```diff
@@ -87,17 +87,24 @@ const Preferences = ({ preferences, setPreferences }: PreferencesProps) => {
   }, [dragOrder])
 
   // Fetch and filter all preferences
-  const { data: preferencesData = [] } = usePreferenceList()
+  const { data: preferencesData = [], error: preferencesError } = usePreferenceList()
   useEffect(() => {
-    setUniquePreferences(
-      preferencesData.reduce(
-        (items, item) =>
-          items.find((x) => x.description === item.description) ? [...items] : [...items, item],
-        []
+    if (!preferencesError) {
```

Comment:
nit: if you did

```
  const { data: preferencesData = [], error: preferencesError } = usePreferenceList()
  // Bail early if there's an error fetching preferenes.
  if (preferencesError) {
    return null
  }
```

you could avoid having to do this additional `!preferencesError` check, yeah?

Reply:
Not exactly, because `useEffect` is called whenever `preferencesData` changes (it's not done serially after the line above it). `preferencesError` is an async value and can change, so we don't really know at any given time what the value might be. So the check inside `useEffect` is necessary no matter where the early return falls.

Reply:
Ah gotcha, thanks for explaining!